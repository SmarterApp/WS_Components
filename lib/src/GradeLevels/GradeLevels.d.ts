export declare enum GradeLevels {
    NA = 0,
    Grade3 = 1,
    Grade4 = 2,
    Grade5 = 4,
    Grade6 = 8,
    Grade7 = 16,
    Grade8 = 32,
    Grade9 = 64,
    Grade10 = 128,
    Grade11 = 256,
    Grade12 = 512,
    Elementary = 7,
    Middle = 56,
    High = 960,
    All = 1023,
}
export declare class GradeLevel {
    static gradeCaseToString(grade: GradeLevels): string | undefined;
    static gradeCaseToShortString(grade: GradeLevels): string | undefined;
    static gradeLevelToString(grades: GradeLevels): string | undefined;
    static gradeLevelContains(haystack: GradeLevels, needle: GradeLevels): boolean;
    static stringToGradeLevel(gradeString: string): GradeLevels;
    static gradeLevelAdd(grade: GradeLevels, gradeString: string): number;
}
