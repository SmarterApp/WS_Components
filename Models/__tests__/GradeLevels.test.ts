import * as GradeLevels from '../GradeLevels';

describe("Grade Levels", () => {
    it("returns each individual string correctly", () => {
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.NA)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade3)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade4)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade5)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade6)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade7)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade8)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade9)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade10)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade11)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Grade12)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Elementary)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.Middle)).toMatchSnapshot();
        expect(GradeLevels.caseToString(GradeLevels.GradeLevels.High)).toMatchSnapshot();
    })

    it("toString() returns a comma delineated string", () => {
        expect(GradeLevels.toString(GradeLevels.GradeLevels.All)).toMatchSnapshot();
    })
})