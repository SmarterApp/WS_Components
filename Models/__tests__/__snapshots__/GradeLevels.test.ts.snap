// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grade Levels caseTostring () returns strings correctly 1`] = `"NA"`;

exports[`Grade Levels caseTostring () returns strings correctly 2`] = `"Grade 3"`;

exports[`Grade Levels caseTostring () returns strings correctly 3`] = `"Grade 4"`;

exports[`Grade Levels caseTostring () returns strings correctly 4`] = `"Grade 5"`;

exports[`Grade Levels caseTostring () returns strings correctly 5`] = `"Grade 6"`;

exports[`Grade Levels caseTostring () returns strings correctly 6`] = `"Grade 7"`;

exports[`Grade Levels caseTostring () returns strings correctly 7`] = `"Grade 8"`;

exports[`Grade Levels caseTostring () returns strings correctly 8`] = `"Grade 9"`;

exports[`Grade Levels caseTostring () returns strings correctly 9`] = `"Grade 10"`;

exports[`Grade Levels caseTostring () returns strings correctly 10`] = `"Grade 11"`;

exports[`Grade Levels caseTostring () returns strings correctly 11`] = `"Grade 12"`;

exports[`Grade Levels caseTostring () returns strings correctly 12`] = `"Elementary"`;

exports[`Grade Levels caseTostring () returns strings correctly 13`] = `"Middle"`;

exports[`Grade Levels caseTostring () returns strings correctly 14`] = `"High"`;

exports[`Grade Levels toString() returns a comma delineated string 1`] = `"Grade 3, Grade 4, Grade 5, Grade 6, Grade 7, Grade 8, Grade 9, Grade 10, Grade 11, Grade 12"`;
